### ISSUE TITLE
Put Twisted developer targetted documentation back on the website

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2459]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#2459|
|Type|enhancement|
|Created|2007-02-19 15:29:07Z|
|Branch|https://github.com/twisted/twisted/tree/devdocs-2459|

All the docs which were moved to doc/development/ no longer appear anywhere on the website.  Several people have been confused/upset to discover this.

<details><summary>Searchable metadata</summary>

```
trac-id__2459 2459
type__enhancement enhancement
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_devdocs_2459 branches/devdocs-2459
branch_author__glyph glyph
status__closed closed
resolution__fixed fixed
component__website website
keywords__documentation documentation
time__1171898947000000 1171898947000000
changetime__1244718600000000 1244718600000000
version__None None
owner__ 
cc__thijs cc__spiv
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_1"></a>|@thijstriemstra commented|
|-|-|

Is this still an issue?(!)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

Yes.  The contents of [doc/development/](doc/development/) still appear nowhere on the website outside of the repository browser.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_3"></a>|@thijstriemstra commented|
|-|-|

Replying to [exarkun](#note_2):
> Yes.  The contents of [doc/development/](doc/development/) still appear nowhere on the website outside of the repository browser.

I'm not able to fix it but raising the priority.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_4"></a>|@thijstriemstra set owner to @thijstriemstra<br>@thijstriemstra set status to `assigned`|
|-|-|

I'll add it to the wiki on TwistedDevelopment

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_5"></a>|@thijstriemstra removed owner<br>@thijstriemstra set status to `new`|
|-|-|

See http://twistedmatrix.com/trac/wiki/TwistedDevelopment?action=diff&version=31

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_6"></a>|@glyph set owner to @thijstriemstra|
|-|-|

Thanks, thijs, this is definitely progress - but it's not really a complete fix.

To be complete, the generated files (i.e. the .html files produced from these .xhtml files by lore) should be on the website somewhere, ideally in the same directory structure as the "howto" documentation.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_7"></a>|@glyph commented|
|-|-|

Whoops, added this to the description when I meant to add it in the comment:

It should probably be possible for you to fix this.  It may not currently be, but can you explain why not, so you can fix other lore-documentation issues?

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_8"></a>|@thijstriemstra commented|
|-|-|

glyph, I'm not sure how and where the Apache configs are stored etc. This seems more like a task for the website maintainer that has access to the server configuration (and root access etc). Or maybe I'm missing something here?

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_9"></a>|@exarkun set owner to @radix|
|-|-|

radix has the access and knowledge required.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_10"></a>|@therve commented|
|-|-|

[#3555](https://github.com/twisted/twisted/issues/3555) has been closed as a duplicate of that, talking about doc/development/legal/contributor-agreement.xhtml.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_12"></a>|@glyph commented|
|-|-|

The reason that this documentation isn't present on the website is that the script responsible for putting stuff on the website, [admin/update-documentation.py](http://twistedmatrix.com/trac/browser/trunk/admin/update-documentation.py?rev=26653), categorizes the documentation by subproject, and 'development' is not a subproject.

This problem was caused by the solution to ticket [#11137](https://github.com/twisted/twisted/issues/11137), which created this new directory.  While separating out development documentation into its own category is a good idea, doing it in this particular way is problematic.  This also means that the development documentation is not released as part of any [ticket:2372 dot product].

I therefore propose that we move 'development' to be a subdirectory of `doc/core` so that it is released with twisted core and put onto the website into the core category.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_13"></a>|@glyph commented|
|-|-|

(In [26950]) Branching to 'devdocs-2459'

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_14"></a>|@glyph set owner to @glyph|
|-|-|

[build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/devdocs-2459)

Moved the development directory into core and added a link to it from the core index.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)|@glyph removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_16"></a>|@therve set owner to @glyph|
|-|-|

Seems correct, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_17"></a>|@glyph set status to `closed`|
|-|-|

(In [27002]) Make developer documentation subsidiary to the 'core' project, instead of off in its own weird directory.

Author: glyph

Reviewer: therve

Fixes [#2459](https://github.com/twisted/twisted/issues/2459)

Aside from just making the structure of the doc/ directory that much more normal, this will have two important effects:

    * the developer documentation will be present in the `TwistedCore` 9.0 release, and
    * the develpoer documentation will be accessible via the website after the 9.0 release.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
